embrio-async/macros/src/lib.rs

```diff
@@ -110,14 +112,92 @@ pub fn async_fn(
 }
 
 fn async_fn_impl(mut item: ItemFn) -> TokenStream {
-    syn::visit_mut::visit_item_fn_mut(
-        &mut AsyncFnTransform::default(),
-        &mut item,
-    );
+    if item.asyncness.is_some() {
+        item.asyncness = None;
+        syn::visit_mut::visit_item_fn_mut(
+            &mut AsyncFnTransform::default(),
+            &mut item,
+        );
+    }
+
+    syn::visit_mut::visit_block_mut(&mut AsyncBlockTransform, &mut item.block);
+    syn::visit_mut::visit_block_mut(&mut ExpandAwait, &mut item.block);
 
     quote!(#item)
 }
 
+struct ExpandAwait;
+
+impl syn::visit_mut::VisitMut for ExpandAwait {
+    fn visit_expr_mut(&mut self, node: &mut syn::Expr) {
+        syn::visit_mut::visit_expr_mut(self, node);
+        let base = match node {
+            syn::Expr::Field(ExprField { member, base, .. }) => {
```

Comment:
This might be possible as something like

```rust
syn::Expr::Field(ExprField { member: Member::Named(member), base, .. }) if member == "await" =>
```

Reply:
Ah, I tried to get something like that working, but didn't have internet access at the time and gave up. I'll give that a shot.